modules/dnsmasq_dhcp/autodetection.go

```diff
@@ -46,21 +60,96 @@ func (d *DnsmasqDHCP) findDHCPRanges() ([]string, error) {
 	return ranges, nil
 }
 
-func findConfigurations(confDir string) ([]string, error) {
-	fis, err := ioutil.ReadDir(confDir)
+// findConfigs recursively finds and reads configuration files respecting
+// conf-file and conf-dir directives.
+// findConfigs is tolerant to IO errors and finds as maximum config
+// files as possible, therefore if an error occurrs during scanning process,
+// it will be just logged with warning severity
+func (d *DnsmasqDHCP) findConfigs(confPath string) []string {
```

Comment:
`findConfigs` returns configs, right? but it doesnt include entry config (`confPath string`). Looks a little bit lame to me, not sure. What do you think?

having it this way we need to

```
	configs := d.findConfigs(d.ConfPath)
	if err != nil {
		d.Warningf("error during configuration dir scanning : %v", err)
    }
	configs = append([]string{d.ConfPath}, configs...)
```

^^ looks like a workaround or smth

Reply:
Didn't touch that logic though

https://github.com/netdata/go.d.plugin/pull/239/files#diff-daece282ff93570d72a99d7ffa78dfd5L20

Reply:
i know, i know.

it was

```
findConfigurations(confDir string) ([]string, error)
```

which is ok (but not good ofc), if finds configs in dir.

we can fix it later.

One things i noticed, your implementation ignores `ConfDir`.

Reply:
Well, maybe we need to get rid of `ConfigDir`

All we need is the entry config

```
CONFIG FILE
       At startup, dnsmasq reads /etc/dnsmasq.conf, if it exists. (On FreeBSD, the file is /usr/local/etc/dnsmasq.conf ) (but see the -C and -7 options.) The format of this file consists of one option per line, exactly as the long options detailed in the OPTIONS section but without the
       leading "--". Lines starting with # are comments and ignored. For options which may only be specified once, the configuration file overrides the command line.  Quoting is allowed in a config file: between " quotes the special meanings of ,:. and # are removed and  the  following
       escapes are allowed: \\ \" \t \e \b \r and \n. The later corresponding to tab, escape, backspace, return and newline.
```

Reply:
On debian it reads `/etc/default/dnsmasq`


from
> /etc/init.d/dnsmasq

```
NAME=dnsmasq
DESC="DNS forwarder and DHCP server"

# Most configuration options in /etc/default/dnsmasq are deprecated
# but still honoured.
ENABLED=1
if [ -r /etc/default/$NAME ]; then
        . /etc/default/$NAME
fi

```

Reply:
ok, it seems `/etc/default/dnsmasq` doesnt work, i found in README in `dnsmasq.d`

```
# All files in this directory will be read by dnsmasq as 
# configuration files, except if their names end in 
# ".dpkg-dist",".dpkg-old" or ".dpkg-new"
#
# This can be changed by editing /etc/default/dnsmasq
```

but, as i said it doesnt work, so we can ignore it.

Reply:
i finally read `man dnsmasq` lmao

```
       -C, --conf-file=<file>
              Specify a different configuration file. The conf-file option is also allowed in configuration files, to include multiple configuration files. A filename of "-" causes dnsmasq to read configuration from stdin.

       -7, --conf-dir=<directory>[,<file-extension>......],
              Read all the files in the given directory as configuration files. If extension(s) are given, any files which end in those extensions are skipped. Any files whose names end in ~ or start with . or start and end with # are always skipped. If the extension starts with * then
              only files which have that extension are loaded. So --conf-dir=/path/to/dir,*.conf loads all files with the suffix .conf in /path/to/dir. This flag may be given on the command line or in a configuration file. If giving it on the command line, be sure to escape  *  charac‐
              ters.

```

it looks like we need both `config_path` and `config_dir` since both of them can be specified via command line.